### DISCUSSION TITLE
Motif enrichment in a set of peaks

### DISCUSSION THREAD

**Discussion by AnjaliC4** [GreenleafLab/ArchR#1331] [Questions / Documentation] [answered]
Hi @rcorces, 

I know we can do motif enrichment in markerPeaks object obtained from ArchR which is a SummarizedExperiment object. If we have done a differential analysis of marker peaks elsewhere - we usually get a data.frame with a set of peaks and foldchange and FDR values. How do I convert this into a format similar to the SummarizedExperiment object accepted by peakAnnoEnrichment() for performing motif enrichment? Please see that the peaks were called with ArchR so are present in the peak matrix.

Thanks

**Comment by rcorces** [comment]
ArchR doesnt have a function to do this (maybe it should). But this can be accomplished with other packages like motifmatchR

**Comment by Zepeng-Mu** [comment, marked as answer]
I happen to be doing something similar, and I construct my own sce from the peaks I have and use this in `peakAnnoEnrichment`, I think this should work:
```r
mySe <- SummarizedExperiment(
  assays = list(z = de_res$z, FDR = de_res$FDR, logFC = de_res$logFC),
  rowData = as.data.frame(peakSet),
  metadata = list(Params = list(useMatrix = "PeakMatrix"))
)

enrichMotifs <- peakAnnoEnrichment(
  seMarker = mySe,
  ArchRProj = projPBMC,
  peakAnnotation = "CisBP2021",
  cutOff = "FDR < 0.1"
)
```

**Reply by AnjaliC4** [reply]
Hi @Zepeng-Mu, Does the above strategy work for you? I think seMarker has to have all the peaks present in project peakSet for this to work?

**Reply by Zepeng-Mu** [reply]
This works for me. For the example above, I also have all the peaks. I think this is mainly because when you construct an `sce` object, you have to include matrices in `assays`, and obviously matrices needs to have the same shape, hence all the peaks, and then you filter peaks in each condition (column) of the `sce` by `cutOff` in `peakAnnoEnrichment`.

I also have cases where the candidate peaks are just a subset of all peaks in the peakSet, in this case if you just have one condition I think you can just use a one-column matrix in `assays`, although it's not all the peaks?
In my case, I constructed a new matrix (`peakBinary` for example) where rows are all peaks and columns are conditions, and then I use 1/0 in the matrix to indicate if the peaks belong to my candidate peak set in each condition. And then I would use something like `peakBinary == 1` in `cutOff` to test for enrichment on those candidate peaks.

Does this answer your question?
